apps/coscrad-frontend/src/store/slices/resources/shared/connected-resources/connected-resources-panel.presenter.tsx

```diff
@@ -1,21 +1,69 @@
+import { ResourceType } from '@coscrad/api-interfaces';
+import { SelectedResourceContainer } from '../../../../../components/higher-order-components/selected-resources.container';
+import { thumbnailResourceDetailPresenterFactory } from '../../../../../components/resources/factories/thumbnail-resource-detail-presenter-factory';
 import { ConnectedResource } from '../../../notes/hooks';
-import { ResourceDetailThumbnail } from './resource-detail-thumbnail';
+
+type SelectedResourcesMap = Map<ResourceType, string[]>;
 
 interface Props {
     data: ConnectedResource[];
 }
 
+/**
+ * Strictly speaking, this is not a presenter. It is a nested container.
+ * We rely on
+ * - first fetching the resource
+ * - then given the resource, fetching its connections
+ * - for each connected resource type, fetching the given resources in the `SelectedResourcesContainer`
+ */
 export const ConnectedResourcesPanelPresenter = ({
     data: connectedResources,
-}: Props): JSX.Element => (
-    <div>
-        <h2>Connected Resources</h2>
-        {connectedResources.map(({ compositeIdentifier, selfContext, otherContext }) => (
-            <div>
-                <ResourceDetailThumbnail {...compositeIdentifier} />
-                self context: {JSON.stringify(selfContext)}
-                other context: {JSON.stringify(otherContext)}
-            </div>
-        ))}
-    </div>
-);
+}: Props): JSX.Element => {
+    /**
+     * We don't necessarily need to show resources of every resource type in
+     * this panel. So we make sure to create selected resources views for only
+     * the resource types we need, so not to needlessly trigger fetches. We
+     * are lazy-loading resources here, on a "fetch all thefirst time the resource
+     * type is seen" basis.
+     */
+    const relevantResourceTypesWithDuplicates = connectedResources.map(
+        ({ compositeIdentifier: { type } }) => type
+    );
+
+    // Remove duplicates
+    const uniqueResourceTypes = [...new Set(relevantResourceTypesWithDuplicates)];
+
+    /**
+     * We initialize an empty map so we don't have to clutter the reducer below
+     * with the "set if not has" logic.
+     */
+    const emptyMap: SelectedResourcesMap = uniqueResourceTypes.reduce(
+        (accMap, resourceType) => accMap.set(resourceType, []),
+        new Map()
+    );
+
+    const resourceTypeToCompositeIds: SelectedResourcesMap = connectedResources.reduce(
+        (acc: SelectedResourcesMap, { compositeIdentifier: { type: resourceType, id } }) =>
+            acc.set(resourceType, [...acc.get(resourceType), id]),
+        emptyMap
+    );
+
+    return (
+        // TODO remove magic string
```

Comment:
On second thought, this is isolated and is a single source of truth. We will not import it from here into tests, the only other place it is referenced, because we do not want to couple tests to implementation. Therefore, this literal, in-line value is ideal.